decoder/src/lib.rs

```diff
@@ -10,6 +10,7 @@
 #![cfg_attr(docsrs, doc(cfg(unstable)))]
 #![doc(html_logo_url = "https://knurling.ferrous-systems.com/knurling_logo_light_text.svg")]
 
+pub const DEFMT_VERSION_3: &str = "3";
 pub const DEFMT_VERSION: &str = "4";
```

Comment:
I think I would do:

```suggestion
pub static DEFMT_VERSIONS: &[&'static str] = ["3", "4"];
```

Reply:
I intentionally didn't include any change that could be viewed as a kind of design decision regarding the way to support multiple protocol versions.
That doesn't mean that I'd be opposed to such a change. I just wanted to leave such decisions to the Ferrous developers. I won't be able to work on this for the next few hours. Feel free to add any changes you like to the pull request.

Reply:
@jonathanpallant, would you prefer this variant?
https://github.com/jannic/defmt/commit/b133f771449a8bc050281ae68f499e16deee9189

Reply:
I would, but let's ask Jorge and Johann.